web/impact/impact/views/algolia_api_key_view.py

```diff
@@ -19,23 +21,38 @@ class AlgoliaApiKeyView(APIView):
     actions = ["GET"]
 
     def get(self, request, format=None):
+        index_prefix = settings.ALGOLIA_INDEX_PREFIX
         client = algoliasearch.Client(
             settings.ALGOLIA_APPLICATION_ID,
             settings.ALGOLIA_API_KEY)
         params = {
             'hitsPerPage': 24,
-            'filters': '',
             'validUntil': int(time.time()) + 3600,
             'userToken': request.user.id
         }
         search_key = settings.ALGOLIA_SEARCH_ONLY_API_KEY
         if request.user.is_staff:
             search_key = settings.ALGOLIA_STAFF_SEARCH_ONLY_API_KEY
-
+        else:
+            active_roles = UserRole.FINALIST_USER_ROLES
+            active_roles.append(UserRole.MENTOR)
+            facet_filters = []
+            for grant in request.user.programrolegrant_set.filter(
+                    program_role__program__program_status__in=CURRENT_STATUSES,
+                    program_role__user_role__name__in=active_roles
```

Comment:
More of a product question - what do mentors see when they access this page? And what do we want them to see?

Reply:
mentors will also see other mentors (confirmed for their program). if we want to restrict it that would be more of a product question